### PR TITLE
All: fix minor warning

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

File: libraries/AP_Math/polygon.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ bool Polygon_outside(const Vector2<T> &P, const Vector2<T> *V, unsigned n)
 template <typename T>
 bool Polygon_complete(const Vector2<T> *V, unsigned n)
 {
-    return (n >= 4 && V[n-1].x == V[0].x && V[n-1].y == V[0].y);
+    return (n >= 4 && V[n-1] == V[0]);
```
Comment (magicrub) [inline libraries/AP_Math/polygon.cpp]: Looks like this change is killing the build. There is no template for vector2<int>

Comment (magicrub) [inline libraries/AP_Math/polygon.cpp]: Vector 2 int.

Comment (khancyr) [inline libraries/AP_Math/polygon.cpp]: Strange

Comment (khancyr) [inline libraries/AP_Math/polygon.cpp]: == operator is define for the template (https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/vector2.cpp#L115) and is_equal is also define for int (https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/AP_Math.cpp#L5)

Comment (khancyr) [inline libraries/AP_Math/polygon.cpp]: OK got it , here is here . not define for int ... https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/vector2.cpp#L153 .
Then I will switch back to manual use of is_equal :-)

**Comment by magicrub** [comment]
I think simply adding a template case for the ==int is best solution.

**Comment by khancyr** [comment]
@magicrub Ok. I add == operator for vector2<int> . Let's wait travis check all device !

**Comment by magicrub** [comment]
Travis likes it. Merged, thanks!

**Comment by rmackay9** [comment]
very nice to have these compiler warnings go away.

**Pull request merged**